OptimizelySDK/IOptimizely.cs

```diff
@@ -41,7 +41,7 @@ public interface IOptimizely
 		/// <param name="userId">The user ID to be used for bucketing.</param>
 		/// <param name="userAttributes">The user's attributes</param>
 		/// <returns>OptimizelyUserContext | An OptimizelyUserContext associated with this OptimizelyClient.</returns>
-		OptimizelyUserContext CreateUserContext(string userId, UserAttributes userAttributes = null);
+		IOptimizelyUserContext CreateUserContext(string userId, UserAttributes userAttributes = null);
```

Comment:
this will break existing functionality. 
e.g. someone used like this.
`OptimizelyUserContext ouc = obj.CreateUserContext(...)
/// value can't be set when returning IOptimizelyUserContext.
`
We should return to OptimizelyUserContext.

Reply:
Well, the idea of the PR is to get the interface, as using the explicit type when declaring is in disuse nowadays. Usually code analyzers mark this as warnings. EG:
- https://www.jetbrains.com/help/resharper/Using_var_Keyword_in_Declarations.html
- https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008

Then, is just a matter of replacing:
`var ouc = obj.CreateUserContext(...);`

I think that, the benefits of **being able to write unit tests** is by far higher for the majority of developers, than the cost of replacing the explicit type by _var_.

What to you thnik?.

Thanks

Reply:
In fact, if the coding standards don't allow the usage of _var_, is just a matter of using the interface instead:

`IOptimizelyUserContext ouc = obj.CreateUserContext(...) `

Reply:
> In fact, if the coding standards don't allow the usage of _var_, is just a matter of using the interface instead:
> 
> `IOptimizelyUserContext ouc = obj.CreateUserContext(...) `

Unfortunately it's not just a matter of coding standards, this would technically require a breaking change to implement in the current version.

We agree this is a worthwhile enhancement for the next major version of the SDK and will add it to our backlog for when that next release is planned.

What we can do in the interim is provide a virtual that will get the user context provided, which can then be overridden for your testing purposes.

Let us know if you think that would be a good stop gap until we can provide the interface change.

Reply:
Yes indeed, this is a breaking change, if you guys consider it useful and want to include this on the next major release, is perfect. 
Then, we can wait, I prefer to close the PR and wait.
Many thanks.

Reply:
But well, you are right, making them virtual is just a nice quick win, and that will allow the mocking library, to intercept the calls and set expectations.

Reply:
I created the PR with the virtual keyword. It's here: https://github.com/optimizely/csharp-sdk/pull/280

It also includes the fix for unit tests running on non English systems.

I'll then, close this PR.

Many thanks!.